File: Cryptography and security libraries and algorithms/C/Code complete/Digital signatures.c
#include <openssl/dsa.h>

DSA *dsa;
// Generate a DSA key
Mykey = DSA_generate_parameters(1024, NULL, 0, NULL, NULL, &ret_code);
if (Mykey == NULL) {
    printf("Error %d in DSA_generate_parameters

File: Cryptography and security libraries and algorithms/Python/Code complete/Digital signatures.py
from cryptography.hazmat.primitives.asymmetric import dsa

# Generate a DSA key
Mykey = dsa.generate_private_key(key_size=1024, backend=default_backend())
Mykey = Mykey.private_bytes(encoding=Encoding.PEM,format=PrivateFormat.PKCS8,encryption_algorithm=NoEncryption())
print('